<launch>
    <arg name="robot_name" default="scout1" doc="Robot name [scout1, scout2, scout3]"/>
    <arg name="use_sim" default="true"/>
    <arg name="paused" default="false"/>

    <rosparam file="$(find sct_gazebo)/config/sct_robot_hw_sim.yaml" command="load" if="$(arg use_sim)"/>

    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="robot_description"
           command="$(find xacro)/xacro $(find sct_description)/urdf/scout/scout_mini.urdf.xacro use_sim:=true "/>

    <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="paused" value="$(arg paused)"/>
        <arg name="world_name" value="$(find sct_gazebo)/worlds/empty.world"/>
    </include>

    <!-- push robot_description to factory and spawn robot in gazebo -->
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" clear_params="true"
          args="-z 0.5 -param robot_description -urdf -model $(arg robot_name)" output="screen"/>

</launch>
